scripts/main/header.js

```diff
@@ -279,11 +279,21 @@ header.setMode = function(mode) {
 				tabindex.makeUnfocusable(e);
 			}
 
-			if (albumID==='s' || albumID==='f' || albumID==='r') {
-				$('#button_info_album, #button_trash_album, #button_visibility_album, #button_move_album').hide();
-				$('.button_add, .header__divider', '.header__toolbar--album').show();
-				tabindex.makeFocusable($('.button_add, .header__divider', '.header__toolbar--album'));
-				tabindex.makeUnfocusable($('#button_info_album, #button_trash_album, #button_visibility_album, #button_move_album'));
+			if (albumID==='starred' || albumID==='unsorted' || albumID === 'public' || albumID==='recent' || album.isTagAlbum()) {
```

Comment:
I'm guessing that in addition to adding support for tag albums you were fixing "regular" smart album support here? I'm confused though because `#button_visibility_album` is now unconditionally shown (in the next line) whereas before it was hidden. Also, you included `unsorted` in this `if` whereas before it was handled in the next `else if (albumID==='0')`, which you've kept (see below), but which will now be unreachable. So something's definitely not right here...

I'm a little surprised by the overall approach here. Why didn't you simply replace the broken `'s'`, `'f'`, and `'r'` here (and `'0'` below) with `'starred'`, `'public'`, `'recent'`, and `'unsorted'`, keeping the insides of those `if`s as-is, and add a new condition just for tag albums further down?

Reply:
Ok I did not understand that albumID=='unsorted' is same as albumID=='0' and I thought that unsorted album should have same funcions as smart albums (but it is not true). So I agree that unsorted shouldn't be in the first condition and I already removed it and changed that zero to unsorted.

Second point is true too. At first I thought that tag albums should haś same behaviour as other smart albums. But it is not true, for example in trash button it is different. So your idea is better, I will refactor it.. thanks for it!

Reply:
But is it ok, that on smart album is visible add (+) button ?

Reply:
I had to fix the publicMode case (we were still displaying the visibility and trash icons).

Answering your question, that's how it's always been. I think we should restrict it (photo uploads to smart albums make some limited sense; subalbum creation definitely not). But let's deal with that in a separate issue; otherwise we'll never get yours out the door! :smile: